lib/gen_stage.ex

```diff
@@ -371,6 +371,29 @@ defmodule GenStage do
     {:stop, reason, new_state} when new_state: term, reason: term, event: term
 
   @doc """
+  Used by :consumer types.
+
+  The subscription is setup to automatically handle demand with
+  `{:automatic, new_state}`, otherwise the callback module must manually send
+  demand by returning `{:manual, new_state}` and calling `ask/2`. When a
+  temporary manual subscription is cancelled `handle_cancel/3` is called.
+  """
+  @callback handle_subscribe(opts :: [options], GenServer.from, state :: term) ::
+    {:automatic | :manual, new_state} |
+    {:automatic | :manual, new_state, timeout | :hibernate} |
+    {:stop, reason, new_state} when new_state: term, reason: term
+
+  @doc """
+  Use by :consumer types
+
+  Handle the cleanup when a temporary manual subscription is cancelled
+  """
+  @callback handle_cancel(cancel_reason :: term, GenServer.from, state :: term) ::
```

Comment:
This `@callback` is not aligned with the implementation. I think we should fix the implementation though and just use the shared callback code (handle_callback or something like that).